# Graded assignment 1

Thijs Verbraeken #2014059

# Packages

We need the package `tidyverse`

In [23]:
library(tidyverse) 

# Assignment 1

First, we generate some random data:

In [24]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2        V3        V4        V5        V6        V7       
1 0.9357330 0.8947530 0.9217016 0.9481127 0.5879141 0.7654191 0.1938863
2 0.3464113 0.7913659 0.5666529 0.9071770 0.3074359 0.3674598 0.9035329
  V8        V9        V10       V11       V12       V13       V14      
1 0.2322676 0.5928495 0.3546888 0.1406745 0.3989030 0.1057565 0.6926828
2 0.3096054 0.9142750 0.1330988 0.1541470 0.5978262 0.9464128 0.1349727
  V15       V16       V17       V18        V19        V20      
1 0.6952579 0.9034754 0.1296182 0.02721123 0.61950097 0.8007899
2 0.6293206 0.7227153 0.7876860 0.02539410 0.07908372 0.9333482

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [30]:
df2 <- df1 %>%
       gather(Key, Value, V1:V20)
head(df2)

Key Value    
1 V1  0.9357330
2 V1  0.3464113
3 V2  0.8947530
4 V2  0.7913659
5 V3  0.9217016
6 V3  0.5666529

## Assignment 2

We will generate some data:

In [26]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [27]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [29]:
df5 <- full_join(df3, df4, by= 'id')
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook